lib/boot.js

```diff
@@ -1,3 +1,19 @@
+const foundElectron = window.process && window.process.type;
+
+if (!foundElectron) {
+  if (localStorage.getItem('appOpened')) {
+    const message = document.createElement('h1');
+    message.innerText =
+      'Simplenote cannot currently be opened simultaneously in more than one tab or window per browser';
+    document.body.appendChild(message);
+    throw new Error('Simplenote can only be opened in one tab');
+  }
+  localStorage.setItem('appOpened', true);
+  window.addEventListener('unload', function(event) {
+    localStorage.removeItem('appOpened');
+  });
+}
```

Comment:
I'm not a fan of relying on `unload` behavior. If the browser dies or someone does something funny then we're going to leave stale state in `localStorage`. In order to keep consistency I think we'll probably need to create a time-based heartbeat. This is going to get really wicked really quickly and we're going to suffer when the clock lies.

```js
const HEARTBEAT_DELAY = 1000;
const clientId = uuid4(); // something (pseudo) random
const emptyLock = [ null, -Infinity ];

if ( 'lock-acquired' !== grabSessionLock() ) {
	ReactDOM.render(
		<ModalBootWarning />
		document.getElementById( 'root' )
	);
	throw new Error( 'Simplenote can only be opened in one tab' );
}

loop( () => {
	switch ( grabSessionLock() ) {
		case 'lock-acquired':
			return true; // keep updating the lock and look for other sessions which may have taken it

		default:
			window.alert( "We've detected another session running Simplenote. This shouldn't be. Expect problems." );
			return false; // stop watching - the user can proceed at their own risk
	}
} );

function loop( f, delay ) {
	f() && setTimeout( () => loop( f, delay ), delay );
}

function grabSessionLock( clientId ) {
	const [ lastClient, lastBeat ] = localStorage.getItem( 'session-lock' ) || emptyLock;
	const now = Date.now();

	// easy case - someone else clearly has the lock
  // add some hysteresis to prevent fighting between sessions
	if ( lastClient !== clientId && now - lastBeat < HEARTBEAT_DELAY * 1.5 ) {
		return 'lock-unavailable';
	}

	// maybe nobody clearly has the lock, let's try and set it
	localStorage.setItem( 'session-lock', [ clientId, now ] );

	// hard case - localStorage is shared mutable state across sessions
	const [ thisClient, thisBeat ] = localStorage.getItem( 'session-lock' ) || emptyLock;

	// someone else set localStorage between the previous two lines of code
	if ( ! ( thisClient === clientId && thisBeat === now ) ) {
		return 'lock-unavailable';
	}

	return 'lock-acquired';
}
```